isobus/src/isobus_virtual_terminal_client.cpp

```diff
@@ -2746,6 +2828,206 @@ namespace isobus
 						}
 						break;
 
+						case static_cast<std::uint8_t>(Function::PreferredAssignmentCommand):
+						{
+							if (0x01 == (data[1] & 0x01))
+							{
+								CANStackLogger::CAN_stack_log("[AUX-N]: Preferred Assignment Error - Auxiliary Input Unit(s) (NAME or Model Identification Code) not valid");
+							}
+							if (0x02 == (data[1] & 0x02))
+							{
+								CANStackLogger::CAN_stack_log("[AUX-N]: Preferred Assignment Error - Function Object ID(S) not valid");
+							}
+							if (0x04 == (data[1] & 0x04))
+							{
+								CANStackLogger::CAN_stack_log("[AUX-N]: Preferred Assignment Error - Input Object ID(s) not valid");
+							}
+							if (0x08 == (data[1] & 0x08))
+							{
+								CANStackLogger::CAN_stack_log("[AUX-N]: Preferred Assignment Error - Duplicate Object ID of Auxiliary Function");
+							}
+							if (0x10 == (data[1] & 0x10))
+							{
+								CANStackLogger::CAN_stack_log("[AUX-N]: Preferred Assignment Error - Other");
+							}
+
+							if (0 != data[1])
+							{
+								std::uint16_t faultyObjectID = (static_cast<std::uint16_t>(data[2]) |
+								                                ((static_cast<std::uint16_t>(data[3])) << 8));
+								CANStackLogger::CAN_stack_log("[AUX-N]: Auxiliary Function Object ID of faulty assignment: " + isobus::to_string(faultyObjectID));
+							}
+							if (0 == data[1])
+							{
+								CANStackLogger::CAN_stack_log("[AUX-N]: Preferred Assignment OK");
+								// TODO: load the preferred assignment into parentVT->auxiliaryInputDevices
+							}
+						}
+						break;
+
+						case static_cast<std::uint8_t>(Function::AuxiliaryAssignmentTypeTwoCommand):
+						{
+							std::uint64_t isoName = (static_cast<std::uint64_t>(data[1]) |
+							                         ((static_cast<std::uint64_t>(data[2])) << 8) |
+							                         ((static_cast<std::uint64_t>(data[3])) << 16) |
+							                         ((static_cast<std::uint64_t>(data[4])) << 24) |
+							                         ((static_cast<std::uint64_t>(data[5])) << 32) |
+							                         ((static_cast<std::uint64_t>(data[6])) << 40) |
+							                         ((static_cast<std::uint64_t>(data[7])) << 48) |
+							                         ((static_cast<std::uint64_t>(data[8])) << 56));
+							bool storeAsPreferred = (0x80 == (data[9] & 0x80));
```

Comment:
Since we've got `CAN_DATA_LENGTH <= message->get_data_length()` up at the top, we may want to additionally check the length required by this message within this case statement so we do not crash if someone sends the `Function::AuxiliaryAssignmentTypeTwoCommand` with length 8.